RallyHereIntegration/Source/RallyHereEditor/Private/RallyHereEditorLootWidget.cpp

```diff
@@ -0,0 +1,1123 @@
+
+		if (!Request.AuthContext->IsLoggedIn())
+		{
+			Request.AuthContext->Refresh();
+			return FReply::Handled();
+		}
+
+		auto* Settings = GetDefault<URH_DevIntegrationSettings>();
+		const FRH_DevSandboxConfiguration* SandboxConfig = Settings->GetSandboxConfiguration(FRallyHereEditorModule::Get().GetSandboxId());
+		FGuid::Parse(SandboxConfig->SandboxGuid, Request.SandboxId);
+
+		Request.LootId = Loot.Get()->GetLootId();
+
+		if (!TSandboxDeleteLoot::DoCall(RH_DevAPIs::GetCatalogAPI(), Request, TSandboxDeleteLoot::Delegate::CreateRaw(this, &SRallyHereEditorLootRow::OnCatalogDeleteLootResponse)))
+		{
+			FMessageDialog::Open(EAppMsgType::Ok, LOCTEXT("DeleteLootError", "An Error occured when attempting to delete the loot."));
+		}
+	}
+
+	return FReply::Handled();
+}
+
+void SRallyHereEditorLootRow::OnCatalogDeleteLootResponse(const TSandboxDeleteLoot::Response& Resp)
+{
+	if (Resp.IsSuccessful())
+	{
+		OnLootDeleteDelegate.ExecuteIfBound(Loot.Get()->GetLootId(), Loot);
+	}
+	else
+	{
+		FMessageDialog::Open(EAppMsgType::Ok, LOCTEXT("DeleteLootError", "An Error occured when attempting to delete the loot."));
+	}
+}
+
+FString SRallyHereEditorLootRow::GetLootRowName()
+{
+	FString RowName = "";
+
+	FGuid Id;
+
+	if (LootUpdateRequest.GetItemId(Id))
+	{
+		RowName += "Item: (" + Id.ToString(EGuidFormats::DigitsWithHyphensLower) + ") ";
+	}
+	
+	if (LootUpdateRequest.GetSubVendorId(Id))
+	{
+		RowName += "SubVendor: (" + Id.ToString(EGuidFormats::DigitsWithHyphensLower) + ")";
+	}
+
+	return RowName;
+}
+
+void SRallyHereEditorLootRow::OnBrowseToItemId(const FGuid& ItemId)
+{
+	static const FName nmItemId(TEXT("ItemId"));
+	if (ItemId.IsValid())
+	{
+		// Search for item id in asset manager
+		UAssetManager& AssetManager = UAssetManager::Get();
+		TArray<FSoftObjectPath> AssetArray;
+
+		TArray<FPrimaryAssetTypeInfo> TypeList;
+		AssetManager.GetPrimaryAssetTypeInfoList(TypeList);
+
+		for (const FPrimaryAssetTypeInfo& TypeInfo : TypeList)
+		{
+			AssetManager.GetPrimaryAssetPathList(TypeInfo.PrimaryAssetType, AssetArray);
+		}
+
+		auto* Settings = GetDefault<URH_DevIntegrationSettings>();
+		
+		FAssetData AssetData;
+		for (const FSoftObjectPath& Path : AssetArray)
+		{
+			if (AssetManager.GetAssetDataForPath(Path, AssetData))
+			{
+				FString ItemIdAsString;
+				if (AssetData.GetTagValue(Settings->ItemIdPropertyName, ItemIdAsString))
+				{
+					if (ItemIdAsString.Contains("Id="))
+					{
+						int32 GuidIndex = ItemIdAsString.Find("Id=");
+						FString GuidAsString = ItemIdAsString.RightChop(GuidIndex + 3).Left(32);
+						FGuid AssetItemId;
+						FGuid::Parse(GuidAsString, AssetItemId);
+
+						if (ItemId == AssetItemId)
+						{
+							TArray<FAssetData> SyncAssets;
+							SyncAssets.Add(AssetData);
+							GEditor->SyncBrowserToObjects(SyncAssets);
+							return;
+						}
+					}
+				}
+			}
+		}
+	}
```

Comment:
Maybe add some sort of notice if no item id that matches could be found.  And/or something about how it searches.

Reply:
Yeah, I figured eventually we will have to document all the .hs for this all.